### ISSUE TITLE
Move ajax GET calls to POST

### ISSUE THREAD

**Issue by CDRO** [fabarea/vidi#110]
Hello there,

I recently ran into a huge problem when hardening my system while trying to use vidi in combination with media.

It appeared to me that I was unable to get the best hardening setting because each time a even tried to use options like suhosin.get.XYZ it broke the functionality of vidi in the backend.

I figured out that this happened because vidi in some places makes GET requests which can take up to 4000 (!) characters in the url.

I switched it to POST and also adapted the ViewHelper/Result/ToJsonViewHelper.php in order to be able to read sEcho correctly. I simply switched from GeneralUtility::_GET to GeneralUtility::_GP.

You'll find the fixed in the attachment.

PS: Too long URLs will sooner or later also affect IE users since there is still a hardcoded limit of 2083 characters for the total url length.

Best Regards
Tizian Schmidlin

[fabarea_vidi_fox_too_long_url.zip](https://github.com/fabarea/vidi/files/68531/fabarea_vidi_fox_too_long_url.zip)

**Comment by fabarea** [comment]
What version do you use? I am curious as on TYPO3 7, it has been greatly improved  (and the length of the URL consequently reduced). The problem comes from the jQuery DataTables plugin which generates lengthy URL. Many built-in parameters are useless for Vidi actually but can't be stripped as easily as that. I could make sense to switching to a POST request if we see that in TYPO3 7 the issue persist. BTW, development of suhosin has been stopped to my knowledge and is therefore deprecated.

**Comment by CDRO** [comment]
Happy new year!

I used the current version of vidi from the TYPO3 extension repository.

I think it would make sense in any case to switch to POST, regardless of persisting problems with the url, just keep in mind what I wrote about IE (who will be around for a long time still).

The reason why we use suhosin is that redhat has still no official distribution of PHP greater 5.4 and therefore there is nothing better at hand.

**Comment by c4ll-m3-j4ck** [comment]
Hey Fabien,

I ran into the same Issue on TYPO3 7.6. These long request URIs still cause Problems when trying to edit categories on systems where many categories (in our case about 560) are present.

It can however easily be fixed, by changing the ajax method to POST in Edit.js:340. I tried by changing it manually and it works as expected.

If you find it necessary, I can create a pull request for the issue!

Best wishes,
Michel
